src/view.c

```diff
@@ -67,6 +81,8 @@ view_matches_query(struct view *view, struct view_query *query)
 {
 	bool match = true;
 	bool empty = true;
+	const struct wlr_security_context_v1_state *security_context =
```

Comment:
I haven't analysed this, but suspect `view_matches_query()` is called a lot, thus making anything within it is therefore an expensive operation.

Unless I'm wrong in that, would it make sense to declare + initialize it to NULL at the top; and then set it to `security_context_from_view()` iff `match && (query->sandbox_engine || query->sandbox_app_id)`?

Reply:
Sure, I reduced the scope of the security_context variable to be used only when needed.